crypto/src/obfuscate.rs

```diff
@@ -0,0 +1,99 @@
+use crate::prelude::{ClientId, ConversationId};
+use hex;
+use log::kv::{ToValue, Value};
+use openmls::framing::Sender;
+use openmls::group::QueuedProposal;
+use openmls::prelude::Proposal;
+use sha2::{Digest, Sha256};
+use std::fmt::{Debug, Formatter};
+
+pub(crate) trait Obfuscate {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result;
+}
+
+impl Obfuscate for &ConversationId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &ClientId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &Proposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self {
+            Proposal::Add(_) => "Add",
+            Proposal::Update(_) => "Update",
+            Proposal::Remove(_) => "Remove",
+            Proposal::PreSharedKey(_) => "PreSharedKey",
+            Proposal::ReInit(_) => "ReInit",
+            Proposal::ExternalInit(_) => "ExternalInit",
+            Proposal::AppAck(_) => "AppAck",
+            Proposal::GroupContextExtensions(_) => "GroupContextExtensions",
+        })
+    }
+}
+
+impl Obfuscate for &QueuedProposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self.proposal {
+            Proposal::Add(_) => "Add",
+            Proposal::Update(_) => "Update",
+            Proposal::Remove(_) => "Remove",
+            Proposal::PreSharedKey(_) => "PreSharedKey",
+            Proposal::ReInit(_) => "ReInit",
+            Proposal::ExternalInit(_) => "ExternalInit",
+            Proposal::AppAck(_) => "AppAck",
+            Proposal::GroupContextExtensions(_) => "GroupContextExtensions",
+        })
+    }
+}
+
+impl Obfuscate for &Sender {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Sender::Member(leaf_node_index) => f.write_fmt(format_args!("Member{}", leaf_node_index)),
```

Comment:
The `f.write_fmt(format_args!(...))` style isn't wrong, but I'm curious why not just use the `write!` macro? It expands to essentially this same thing, as I understand it, but it's easier to read.

Like

```rust
write!(f, "Member{leaf_node_index}")
```